src/v/storage/record_batch_builder.cc

```diff
@@ -26,6 +26,8 @@ record_batch_builder::record_batch_builder(
 record_batch_builder::~record_batch_builder() {}
 
 model::record_batch record_batch_builder::build() && {
+    static constexpr auto data_batch_type
+      = model::well_known_record_batch_types[1];
```

Comment:
should we move raft::data_batch_type to a common location now so we don't have to duplicate the bare number?

Reply:
I will do this in separate PR. This commit will be dropped